src/main/java/ResourceDiscovery/ProjectObjects/ProjectAssetsUpdater.java

```diff
@@ -0,0 +1,205 @@
+package ResourceDiscovery.ProjectObjects;
+
+import ResourceDiscovery.*;
+import ResourceDiscovery.AssetObjects.AssetObject;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.common.flogger.FluentLogger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class ProjectAssetsUpdater {
+    private static final String PROJECT_ID_EXP = "{project_id}";
+    private static final String ZONE_NAME_EXP = "{zone_name}";
+    private static final String ASSET_TYPE_EXP = "{asset_type}";
+
+    public enum AssetTypes {INSTANCE_COMPUTE_ASSET, DISK_COMPUTE_ASSET, TOPIC_PUB_SUB_ASSET,
```

Comment:
If these enums are used outside of this class (which they are), it's better to declare them in a separate namespace/class of their own, to which all usages would point to.